MyOwnORM/MyOwnORM/Implementations/CustomDbSetRepository.cs

```diff
@@ -0,0 +1,523 @@
+                    {
+                        if (await reader.ReadAsync())
+                        {
+                            T type = dbSetReflection.MapReaderToEntity(reader);
+                            entity = type;
+                        }
+                    }
+                }
+            }
+            return entity;
+        }
+
+        public async Task InsertAsync(T obj)
+        {
+            using (SqlConnection connection = new SqlConnection(_connectionString))
+            {
+                await connection.OpenAsync();
+
+                string values = reflectionHelper.MapEntityPropertyValuesInString(obj);
+
+                query = $"INSERT INTO {tableName} VALUE ({values})";
+
+                SqlCommand command = new SqlCommand(query, connection);
+                await command.ExecuteNonQueryAsync();
+            }
+        }
+        public async Task InsertCascadeAsync(T obj)
+        {
+            string[] names = dbSetReflection.GetNamesOfCollectionOrModel();
+            using (SqlConnection connection = new SqlConnection(_connectionString))
+            {
+                bool hasRowInDataBase = false;
+                await connection.OpenAsync();
+
+                string values = reflectionHelper.MapEntityPropertyValuesInString(obj, names);
+                string idNameEntity = dbSetReflection.GetIdProperty(obj);
+                string idEntity = dbSetReflection.GetIdPropertyValue(obj);
+
+                string checkQuery = $"SELECT * FROM {typeof(T).Name} WHERE {idNameEntity} = {idEntity}";
+
+                using (SqlCommand commandCheck = new SqlCommand(checkQuery, connection))
+                {
+                    using (SqlDataReader reader = await commandCheck.ExecuteReaderAsync())
+                    {
+                        while (await reader.ReadAsync())
+                        {
+                            object id = reader[$"{idNameEntity}"];
+                            if (id == null)
+                            {
+                                hasRowInDataBase = true;
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                if (!hasRowInDataBase)
+                {
+                    string query = $"INSERT INTO {typeof(T).Name} VALUES ({values})";
+
+                    SqlCommand command = new SqlCommand(query, connection);
+                    await command.ExecuteNonQueryAsync();
+                }
+
+                string[] queries = dbSetReflection.InsertCascadeModelsOrCollection(obj, names);
+
+                for (int i = 0; i < queries.Length; i++)
+                {
+                    SqlCommand commandCascade = new SqlCommand(queries[i], connection);
+                    await commandCascade.ExecuteNonQueryAsync();
+                }
+
+            }
+        }
+        public async Task UpdateAsync(T obj)
```

Comment:
how do you choice update/updatecascade? Similar question to insert and delete

Reply:
When you only need to update an entity with no relationship, you use 'UpdateAsync' and if you have a relationship in an entity, you use cascading methods. I have separated the methods into cascaded and non-cascaded methods because it will increase performance and reduce application load.